src/support/lockedpool.cpp

```diff
@@ -253,8 +253,10 @@ void *PosixLockedPageAllocator::AllocateLocked(size_t len, bool *lockingSuccess)
     }
     if (addr) {
         *lockingSuccess = mlock(addr, len) == 0;
-#ifdef MADV_DONTDUMP
+#if defined(MADV_DONTDUMP) // Linux
         madvise(addr, len, MADV_DONTDUMP);
+#elif defined(MADV_NOCORE) // FreeBSD
```

Comment:
I think it would be preferable to use a new `#ifdef` here, just in case there's ever a system with both defined, but only one supported at runtime...

Reply:
As there are currently no such systems, and the flags do exactly the same thing, it seems unlikely that that would ever be an issue.

Reply:
even more if both defined this is "bizarro" territory and it might be better to fail the compile to reconsider what to do here, i don't think naively calling both in order (in what order?) is ever a good idea …
(but to be clear: leaving it like this is fine with me)

Reply:
I think if there is a system that defines `MADV_DONTDUMP`, but it is not supported at runtime (!?) then that system is badly broken. Regardless if it also defines `MADV_NOCORE` that works.